baseplate/clients/thrift.py

```diff
@@ -167,17 +167,18 @@ def _call_thrift_method(self: Any, *args: Any, **kwargs: Any) -> Any:
         trace_name = f"{self.namespace}.{name}"
         last_error = None
 
-        for time_remaining in self.retry_policy:
-            try:
-                with self.pool.connection() as prot:
-                    span = self.server_span.make_child(trace_name)
-                    span.set_tag("slug", self.namespace)
-
-                    client = self.client_cls(prot)
-                    method = getattr(client, name)
-                    span.set_tag("method", method.__name__)
-                    span.start()
-
+        with self.pool.connection() as prot:
+            span = self.server_span.make_child(trace_name)
+            span.set_tag("protocol", "thrift")
+            span.set_tag("slug", self.namespace)
+
+            client = self.client_cls(prot)
+            method = getattr(client, name)
+            span.set_tag("method", method.__name__)
+            span.start()
```

Comment:
retrying should start a new span, not one span wrapping all the retries. This change would limit the visibility into why a call failed and should have been retried, along with how many retries and how long each one took.

Reply:
OK, moving the `for` loop up before this initialization of the span. Initialization needs to exist outside of the `try` block since  `span` must be assigned and `span.start()` must be called before `span.finish()`. Otherwise every exception is rethrown, either an UnboundLocalError or another one about a timer not being started. 

Additionally all tags must be set before `start()` is called, otherwise they're seemingly just dropped. It worries me a bit since that means we need to call `pool.connection` and `client_cls` outside the `try` block and I'm not sure how effectful those are.